frontend/src/component/environments/EnvironmentTable/EnvironmentActionCell/EnvironmentCloneModal/EnvironmentCloneModal.tsx

```diff
@@ -0,0 +1,385 @@
+        type: 'CLIENT',
+        environment: name,
+        projects: tokenProjects,
+    });
+
+    const handleSubmit = async (e: FormEvent<HTMLFormElement>) => {
+        e.preventDefault();
+
+        try {
+            await cloneEnvironment(
+                environment.name,
+                getCloneEnvironmentPayload()
+            );
+            const response = await createToken(getApiTokenCreatePayload());
+            if (apiTokenGeneration === APITokenGeneration.NOW) {
+                const token = await response.json();
+                newToken(token);
+            }
+            setToastData({
+                title: 'Environment successfully cloned!',
+                type: 'success',
+            });
+            refetchEnvironments();
+            setOpen(false);
+        } catch (error: unknown) {
+            setToastApiError(formatUnknownError(error));
+        }
+    };
+
+    const formatApiCode = () => {
+        return `curl --location --request POST '${
+            uiConfig.unleashUrl
+        }/api/admin/environments/${environment.name}/clone' \\
+    --header 'Authorization: INSERT_API_KEY' \\
+    --header 'Content-Type: application/json' \\
+    --data-raw '${JSON.stringify(getCloneEnvironmentPayload(), undefined, 2)}'`;
+    };
+
+    const isNameNotEmpty = (name: string) => name.length;
+    const isNameUnique = (name: string) =>
+        !environments?.some(environment => environment.name === name);
+    const isValid =
+        isNameNotEmpty(name) && isNameUnique(name) && tokenProjects.length;
+
+    const onSetName = (name: string) => {
+        clearError(ErrorField.NAME);
+        if (!isNameUnique(name)) {
+            setError(
+                ErrorField.NAME,
+                'An environment with that name already exists.'
+            );
+        }
+        setName(name);
+    };
+
+    const selectableProjects = allProjects.map(project => ({
+        value: project.id,
+        label: project.name,
+    }));
+
+    return (
+        <SidebarModal
+            open={open}
+            onClose={() => {
+                setOpen(false);
+            }}
+            label={`Clone ${environment.name} environment`}
+        >
+            <FormTemplate
+                loading={loading}
+                modal
+                title={`Clone ${environment.name} environment`}
+                description="Cloning an environment will clone all feature toggles and their configuration (activation strategies, segments, status, etc) into a new environment."
+                documentationLink="https://docs.getunleash.io/user_guide/environments#cloning-environments"
+                documentationLinkLabel="Cloning environments documentation"
+                formatApiCode={formatApiCode}
+            >
+                <StyledForm onSubmit={handleSubmit}>
+                    <div>
+                        <StyledInputDescription>
+                            What is your environment name? (Can't be changed
```

Comment:
Entirely optional but I think this is a little clearer

```suggestion
                            What is your new environment name? (Can't be changed
```